File: backend/BlockEngine/Application/Plugins/Callout/CalloutBlockContent.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace BlockEngine.Application.Plugins.Callout;

public class CalloutBlockContent
{
    [JsonPropertyName("variant")]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public CalloutVariant Variant { get; set; }
    
    [JsonPropertyName("text")]
    public string Text { get; set; }
}

File: frontend/src/entities/module-block/ui/callout/callout-block-content.ts
export type CalloutVariant = "tip" | "note"

export type CalloutBlockContent = {
    variant: CalloutVariant
    text: string
}

export const CALLOUT_TITLES: Record<CalloutVariant, string> = {
    note: "Важно запомнить",
    tip: "Совет"
}